collector/node_collector_test.go

```diff
@@ -66,26 +66,26 @@ func TestBuildDatapoints(t *testing.T) {
 	})
 }
 
-func TestNewAgent(t *testing.T) {
-	Convey("When establishing a new agentPoller", t, func() {
+func TestNewDCOSHost(t *testing.T) {
+	Convey("When establishing a new DCOSHost object", t, func() {
 		Convey("Should return an error when given an improper ip-detect command", func() {
-			_, err := NewAgent("", 10000, 60, make(chan<- producers.MetricsMessage))
+			_, err := NewDCOSHost("agent", "", 10000, 60, make(chan<- producers.MetricsMessage))
```

Comment:
this is exactly the reason why functional parameters is a great pattern for constructors. Some of the parameters might must have default or even empty values. Here is an example: https://github.com/dcos/dcos-go/blob/29e9e7508543aedf36a8bbccaffdf87d1fbe0f20/exec/exec.go#L64 and https://github.com/dcos/dcos-go/blob/29e9e7508543aedf36a8bbccaffdf87d1fbe0f20/jwt/transport/transport.go#L48